diff -uraN speech-dispatcher-0.12.1/src/audio/alsa.c omv-speech-dispatcher-0.12.1/src/audio/alsa.c
--- speech-dispatcher-0.12.1/src/audio/alsa.c	2025-03-17 10:15:28.000000000 +0100
+++ omv-speech-dispatcher-0.12.1/src/audio/alsa.c	2025-09-20 20:19:59.405537994 +0200
@@ -36,6 +36,7 @@
 
 #include <alsa/asoundlib.h>
 #include <alsa/pcm.h>
+#include "spd_utils.h"
 
 #ifdef USE_DLOPEN
 #define SPD_AUDIO_PLUGIN_ENTRY spd_audio_plugin_get
diff -uraN speech-dispatcher-0.12.1/src/audio/libao.c omv-speech-dispatcher-0.12.1/src/audio/libao.c
--- speech-dispatcher-0.12.1/src/audio/libao.c	2025-03-17 10:15:28.000000000 +0100
+++ omv-speech-dispatcher-0.12.1/src/audio/libao.c	2025-09-20 20:20:26.943790380 +0200
@@ -31,6 +31,7 @@
 #include <string.h>
 #include <glib.h>
 #include <ao/ao.h>
+#include "spd_utils.h"
 
 #ifdef USE_DLOPEN
 #define SPD_AUDIO_PLUGIN_ENTRY spd_audio_plugin_get
diff -uraN speech-dispatcher-0.12.1/src/audio/nas.c omv-speech-dispatcher-0.12.1/src/audio/nas.c
--- speech-dispatcher-0.12.1/src/audio/nas.c	2025-05-06 20:44:34.000000000 +0200
+++ omv-speech-dispatcher-0.12.1/src/audio/nas.c	2025-09-20 20:20:39.559994072 +0200
@@ -31,6 +31,7 @@
 #include <audio/soundlib.h>
 
 #include <pthread.h>
+#include "spd_utils.h"
 
 #ifdef USE_DLOPEN
 #define SPD_AUDIO_PLUGIN_ENTRY spd_audio_plugin_get
diff -uraN speech-dispatcher-0.12.1/src/audio/oss.c omv-speech-dispatcher-0.12.1/src/audio/oss.c
--- speech-dispatcher-0.12.1/src/audio/oss.c	2025-03-17 10:15:28.000000000 +0100
+++ omv-speech-dispatcher-0.12.1/src/audio/oss.c	2025-09-20 20:20:48.756415273 +0200
@@ -38,6 +38,7 @@
 #include <glib.h>
 
 #include <sys/soundcard.h>
+#include "spd_utils.h"
 
 #ifdef USE_DLOPEN
 #define SPD_AUDIO_PLUGIN_ENTRY spd_audio_plugin_get
diff -uraN speech-dispatcher-0.12.1/src/audio/pipewire.c omv-speech-dispatcher-0.12.1/src/audio/pipewire.c
--- speech-dispatcher-0.12.1/src/audio/pipewire.c	2025-03-17 10:15:28.000000000 +0100
+++ omv-speech-dispatcher-0.12.1/src/audio/pipewire.c	2025-09-20 20:20:58.074830198 +0200
@@ -37,6 +37,7 @@
 #include <pthread.h>
 #include <pipewire/loop.h>
 #include <spa/support/system.h>
+#include "spd_utils.h"
 
 // for the buffer size which will be allocated on the heap, to store the samples produced by speech dispatcher and read by pipewire
 //  this buffer will be used as backing storage by a ring buffer, which will also insure that the threads are syncronised when reading from and writing to the storage area
diff -uraN speech-dispatcher-0.12.1/src/audio/pulse.c omv-speech-dispatcher-0.12.1/src/audio/pulse.c
--- speech-dispatcher-0.12.1/src/audio/pulse.c	2025-03-17 10:15:28.000000000 +0100
+++ omv-speech-dispatcher-0.12.1/src/audio/pulse.c	2025-09-20 20:21:05.664354702 +0200
@@ -50,6 +50,7 @@
 #include <pulse/pulseaudio.h>
 #include <pulse/thread-mainloop.h>
 #include <pulse/error.h>
+#include "spd_utils.h"
 
 #ifdef USE_DLOPEN
 #define SPD_AUDIO_PLUGIN_ENTRY spd_audio_plugin_get
